api/database/factories/ModelFactory.php

```diff
@@ -13,16 +13,53 @@
 
 use Carbon\Carbon;
 
+// Ensure that the custom validation rules are registered so the factories also
+// have them available.
+Validator::resolver(function ($translator, $data, $rules, $messages) {
+    return new \App\Services\SpiraValidator($translator, $data, $rules, $messages);
+});
+
+$factory->define(App\Models\TestEntity::class, function ($faker) {
+    return [
+        'entity_id' => $faker->uuid,
+        'varchar' => $faker->word,
+        'hash' => Hash::make($faker->word),
+        'integer' => $faker->numberBetween(0, 500),
+        'decimal' => $faker->randomFloat(2, 0, 100),
+        'boolean' => $faker->boolean(),
+        'nullable' => null,
+        'text' => $faker->paragraph(3),
+        'date' => $faker->date(),
+        'multi_word_column_title' => true,
+        'hidden' => $faker->boolean()
+    ];
+});
+
+$factory->defineAs(App\Models\TestEntity::class, 'custom', function ($faker) use ($factory) {
+    $testEntity = $factory->raw(App\Models\TestEntity::class);
+
+    return array_merge($testEntity, ['varchar' => 'custom']);
+});
+
+
 $factory->define(App\Models\User::class, function ($faker) {
+
+    // Prepare available country codes (Faker does not generate valid country codes)
+    $countries = \App::make('App\Services\Datasets\Countries')
+            ->all()
+            ->toArray();
+    $countryCodes = array_fetch($countries, 'country_code');
```

Comment:
@artstorm by using your implementation in the mock factory your unit tests are almost guaranteed to pass, as a problem in the implementation will become a problem in the mock. It would be better here to just pick a random element from a hardcoded list of known country codes. eg `$countryCodes = ['au', 'nz', 'th', 'se']` etc.